Bottom: 6f33d9c40ca250370d558f0cceb1bc54a947573d
Top:    b2b9689c25487322b85e05295af90cb078314c91
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-09-30 22:52:08 +0100

Refresh of don-t-bother-unsubtracting-com

---

diff --git a/src/polygon.c b/src/polygon.c
index b20c350..c193530 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -65,7 +65,7 @@ RCSID ("$Id$");
 
 #define ROUND(x) ((long)(((x) >= 0 ? (x) + 0.5  : (x) - 0.5)))
 
-#define UNSUBTRACT_BLOAT 50
+#define UNSUBTRACT_BLOAT 10
 
 /* ---------------------------------------------------------------------------
  * local prototypes
@@ -923,8 +923,7 @@ UnsubtractPin (PinType * pin, LayerType * l, PolygonType * p)
     return 0;
   if (!Unsubtract (np, p))
     return 0;
-  clearPoly (PCB->Data, l, p, (const BoxType *) pin,
-             0.1 * (pin->Thickness + pin->Clearance));
+  clearPoly (PCB->Data, l, p, (const BoxType *) pin, 2 * UNSUBTRACT_BLOAT);
   return 1;
 }
 
@@ -943,7 +942,7 @@ UnsubtractArc (ArcType * arc, LayerType * l, PolygonType * p)
     return 0;
   if (!Unsubtract (np, p))
     return 0;
-  clearPoly (PCB->Data, l, p, (const BoxType *) arc, 50);
+  clearPoly (PCB->Data, l, p, (const BoxType *) arc, 2 * UNSUBTRACT_BLOAT);
   return 1;
 }
 
@@ -962,7 +961,7 @@ UnsubtractLine (LineType * line, LayerType * l, PolygonType * p)
     return 0;
   if (!Unsubtract (np, p))
     return 0;
-  clearPoly (PCB->Data, l, p, (const BoxType *) line, 50);
+  clearPoly (PCB->Data, l, p, (const BoxType *) line, 2 * UNSUBTRACT_BLOAT);
   return 1;
 }
 
@@ -981,7 +980,7 @@ UnsubtractText (TextType * text, LayerType * l, PolygonType * p)
     return -1;
   if (!Unsubtract (np, p))
     return 0;
-  clearPoly (PCB->Data, l, p, (const BoxType *) text, 101);
+  clearPoly (PCB->Data, l, p, (const BoxType *) text, 2 * UNSUBTRACT_BLOAT);
   return 1;
 }
 
@@ -997,7 +996,7 @@ UnsubtractPad (PadType * pad, LayerType * l, PolygonType * p)
     return 0;
   if (!Unsubtract (np, p))
     return 0;
-  clearPoly (PCB->Data, l, p, (const BoxType *) pad, 100);
+  clearPoly (PCB->Data, l, p, (const BoxType *) pad, 2 * UNSUBTRACT_BLOAT);
   return 1;
 }
